float64 transX
float64 transY
float64 transZ
float64 rotX
float64 rotY
float64 rotZ 
